### PR TITLE
use Nexus_Memory_Free for Nexus allocated pointers

### PR DESCRIPTION
**Pull request by artur-gebicz-ml**

### CODE REVIEW

File: MediaSystem.cpp [hunk 1]
```diff
@@ -601,12 +601,17 @@ class PlayReady : public IMediaKeys, public IMediaKeysExt {
 
     ErrorExit:
         SAFE_OEM_FREE(pbChallenge);
-        SAFE_OEM_FREE(pTimeChallengeURL);
-        SAFE_OEM_FREE(pbResponse);
+
+        if(pTimeChallengeURL) {
```
Comment (pawel-stanek-ml) [inline MediaSystem.cpp]: pTimeChallengeURL != nullptr by the style

File: MediaSystem.cpp [hunk 2]
```diff
@@ -601,12 +601,17 @@ class PlayReady : public IMediaKeys, public IMediaKeysExt {
...
+            NEXUS_Memory_Free(pTimeChallengeURL);
+        }
+        if(pbResponse) {
```
Comment (pawel-stanek-ml) [inline MediaSystem.cpp]: Ditto.

File: MediaSystem.cpp [hunk 3]
```diff
@@ -601,12 +601,17 @@ class PlayReady : public IMediaKeys, public IMediaKeysExt {
 
     ErrorExit:
         SAFE_OEM_FREE(pbChallenge);
```
Comment (pawel-stanek-ml) [inline MediaSystem.cpp]: Make sure in GenerateChallenge spec it's ALWAYS ok to call OEM_FREE here

File: MediaSystem.cpp (hunk 2 above)
Comment (artur-gebicz-ml) [reply MediaSystem.cpp]: done

File: MediaSystem.cpp (hunk 1 above)
Comment (artur-gebicz-ml) [reply MediaSystem.cpp]: done

File: MediaSystem.cpp [hunk 4]
```diff
@@ -602,10 +602,11 @@ class PlayReady : public IMediaKeys, public IMediaKeysExt {
...
 
-        if(pTimeChallengeURL) {
+        if (pTimeChallengeURL == nullptr) {
```
Comment (pawel-stanek-ml) [inline MediaSystem.cpp]: != nullptr

File: MediaSystem.cpp (hunk 3 above)
Comment (artur-gebicz-ml) [reply MediaSystem.cpp]: As per documentation:
```
The caller must free f_ppbChallenge using Oem_MemFree after use.
```

Comment (pawel-stanek-ml) [reply MediaSystem.cpp]: Cool. We're good then

**Pull request merged**